packages/inertia-vue/src/form.js

```diff
@@ -0,0 +1,99 @@
+import Vue from 'vue'
+import { Inertia } from '@inertiajs/inertia'
+
+export default data => {
+  const form = {
+    data: { ... data },
+    defaults: { ... data },
+    errors: {},
+    processing: false,
+    submit(method, url, options) {
+      url = url || Inertia.page.url
+      options = options || {}
+
+      Inertia[method](url, form.data, {
+        ... options,
+        onStart: visit => {
+          form.processing = true
+
+          if (options.onStart) {
+            return options.onStart(visit)
+          }
+        },
+        onFinish: () => {
+          form.processing = false
+
+          if (options.onFinish) {
+            return options.onFinish()
+          }
+        },
+        onError: errors => {
+          Vue.set(form, 'errors', errors)
+
+          if (options.onError) {
+            return options.onError(errors)
+          }
+        },
+        onSuccess: page => {
+          Vue.set(form, 'errors', {})
+
+          if (options.onSuccess) {
+            return options.onSuccess(page)
+          }
+        },
+      })
+    },
+    post(url, options) {
+      form.submit('post', url, options)
+    },
+    put(url, options) {
+      form.submit('put', url, options)
+    },
+    patch(url, options) {
+      form.submit('patch', url, options)
+    },
+    delete(url, options) {
+      form.submit('delete', url, options)
+    },
+    reset(fields) {
```

Comment:
When resetting the form, should we not also reset the errors to an empty object? It seems to me like if I'm resetting the form, I also want to clear any previous errors as well.

In fact, I'd even suggest having a `form.clearErrors()` method, just in case you want to manually clear the errors.

Reply:
> In fact, I'd even suggest having a form.clearErrors() method, just in case you want to manually clear the errors.
+1 for the `form.clearErrors()`. No concrete use case right now, but seems nice to have.

> It seems to me like if I'm resetting the form, I also want to clear any previous errors as well.
Use cases I can think of:
1. Creating stuff and resetting the form and errors afterwards to add another entry. But in this case the errors are already reset by the `onSuccess` handler.
2. User wants to reset the form. So it would make sense to reset the errors aswell, as they are no longer valid.

Anything else?

Reply:
Yeah, so, while I'm not opposed to this, I'd really like to know of some _real_ use cases first.

1. With Inertia this would already be reset because of the Inertia visit that happens in the background, so it's really not something you would do
2. Would you really do this one? So far with pretty much every webpage I've seen ever, this doesn't happen, and I'm really having a hard time coming up with a real scenario for this?

Reply:
So the use-case I'm thinking is the modal form in my app. When you close the modal, and then open it again, it would be nice to call `form.reset()`, and have all the original form values reset and the errors cleared. 👍

Reply:
Yep, okay, adding it in 👍